test/test_dynamic_shapes.py

```diff
@@ -175,13 +191,173 @@ def test_expand_symint_correctness(self):
     t2 = torch.zeros([size1, size2], device=dev)
     t2[3][0] = 1
     t2[3][1] = 1
-    # t2 has size [<=10, 2]
+    # t3 has size [<=10, 2]
     t3 = torch.nonzero(t2)
     t4 = torch.ones([size1, size2], device=dev)
     expand_out_xla = t4.expand(t3.shape[0], size1, size2)
     self.assertEqual(t3.shape[0], 2)
     self.assertEqual(expand_out_aten.cpu(), expand_out_xla.cpu())
 
+  def test_unsqueeze_copy_dynamism(self):
+    t1 = torch.tensor([[1, 0, 0, 5, 0, 6], [1, 3, 2, 0, 0, 1]], device=dev)
+    t2 = torch.nonzero(t1)
+    # t2.shape=torch.Size([<=12, 2]) with real size [7, 2]
+
+    t2_unsqueeze = torch.unsqueeze(t2, 0)
+
+    self.assertEqual(len(t2_unsqueeze.size()), 3)
+    self.assertIsInstance(t2_unsqueeze.shape[0], int)
+    self.assertIsInstance(t2_unsqueeze.shape[1], torch.SymInt)
+    self.assertIsInstance(t2_unsqueeze.shape[2], int)
+    self.assertEqual(str(t2_unsqueeze.shape[0]), '1')
+    self.assertEqual(str(t2_unsqueeze.shape[1]), '<=12')
+    self.assertEqual(str(t2_unsqueeze.shape[2]), '2')
+    self.assertEqual(t2_unsqueeze.shape[0], 1)
+    self.assertEqual(t2_unsqueeze.shape[1], 7)
```

Comment:
general comment, this line will trigger a device execution. We should have a test doing something similar to 
```
self.assertEqual(t2_unsqueeze.shape[1], 7)
# check execution counter incremented by 1
xm.mark_step()
# check execution counter incremented by 1
self.assertEqual(t2_unsqueeze.shape[1], 7)
# check execution counter does not increase
```
I am actually not sure if `t2_unsqueeze.shape` will get materalized upon a `mark_step`, that;s one thing we want to get some clarity. I am guessing it won't because this dynamic IR is not being assigned to any tensor.. now I am not sure what's going to happen in above case, which is why we need a test.

Reply:
Thanks for the suggestion. I wonder by checking the execution counter, what aspect of the code are we testing?

Reply:
we check that accessing the real dynamic size will trigger a device computation. I was hoping to test that after `mark_step` the `tensor.size` will get also materalize, but I am not sure if that;s the case.

Reply:
Thanks. let me add the test in a followup pr.